app/src/main/java/org/sopt/and/ui/sign/SignInScreen.kt

```diff
@@ -18,158 +15,195 @@ import androidx.compose.material3.ButtonDefaults
+    signIn: Screen.SignIn,
+    navigateToMy: () -> Unit,
+    navigateToSignUp: () -> Unit,
     modifier: Modifier = Modifier,
+    viewModel: SignInViewModel = viewModel(),
 ) {
+    val signInState by viewModel.signInState.collectAsState()
+    val signInSuccess by viewModel.signInSuccess.collectAsState()
+    val context = LocalContext.current
+
+    LaunchedEffect(signInSuccess) {
+        if (signInSuccess) {
+            navigateToMy()
+            viewModel.resetSignInSuccess()
+        }
+    }
+    signInState.snackbarMessage?.let { message ->
+        LaunchedEffect(message) {
+            viewModel.clearSnackbarMessage() // 메시지 초기화
+            CoroutineScope(Dispatchers.Main).launch {
+                SnackBarUtils.showSnackBar(
+                    message = message,
+                    actionLabel = context.getString(R.string.sign_in_snackbar_action_close)
+                )
+            }
+        }
+    }
     Column(
         modifier = modifier
             .fillMaxSize()
             .background(WavveBg)
-            .padding(16.dp)
-    ){
-        Spacer(modifier = Modifier.height(32.dp))
-        WavveCommonTextField(
-            value = email,
-            onValueChange = onEmailChange,
-            hint = stringResource(R.string.sign_in_text_field_id_hint)
-        )
-        Spacer(modifier = Modifier.height(4.dp))
-        WavveCommonPasswordField(
-            value = password,
-            onValueChange = onPasswordChange,
-            hint = stringResource(R.string.sign_in_text_field_password_hint)
-        )
-
-        Spacer(modifier = Modifier.height(24.dp))
-
-        Button(
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(horizontal = 8.dp)
-            ,
-            colors = ButtonDefaults.buttonColors(
-                containerColor = WavvePrimary
-            ),
-            onClick = {
-                onSignInButtonClick()
-            },
+    ) {
+        BackButtonTopBar({ /*TODO : 뒤로가기처리*/ })
+        Column(
+            modifier = modifier
+                .fillMaxSize()
+                .background(WavveBg)
+                .padding(16.dp)
         ) {
-            Text(
-                text = stringResource(R.string.sign_in_text_login),
-                color = Color.White,
-                fontSize = 16.sp,
-                modifier = Modifier.padding(vertical = 8.dp)
-            )
-        }
-
-        Spacer(modifier = Modifier.height(16.dp))
-
-        Row(
-            modifier = Modifier
-                .align(Alignment.CenterHorizontally),
-            horizontalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterHorizontally)
-        ){
-            Text(
-                text = stringResource(R.string.sign_in_text_find_id),
-                fontSize = 12.sp,
-                color = Gray3,
+            WavveCommonTextField(
+                value = signInState.email,
+                onValueChange = { viewModel.updateEmail(it) },
```

Comment:
저는 이런 람다의 경우, it 대신 이름을 붙여주는게 읽기 편하더라구요!
```suggestion
                onValueChange = { newEmail ->
                   viewModel.updateEmail
                },
```

혹은 인자가 없거나 순서가 함수의 매개변수 순서와 같다면 람다를 없애고 참조 연산자`::` 를 사용해도 됩니다~
```suggestion
                onValueChange = viewModel::updateEmail,
```

Reply:
이거 진짜 혁명이다